sync/sync_head.go

```diff
@@ -191,5 +191,8 @@ func isExpired(header header.Header, period time.Duration) bool {
 
 // isRecent checks if header is recent against the given blockTime.
 func isRecent(header header.Header, blockTime time.Duration) bool {
-	return time.Since(header.Time()) <= blockTime+blockTime/2 // add half block time drift
+	// double the block time and add half block time drift to allow headersub
+	// some time to deliver a more recent head before eagerly requesting a new
+	// head
+	return time.Since(header.Time()) <= (blockTime*2)+blockTime/2
```

Comment:
IIRC, we discussed 3-5 secs. We don't lose much if we make it bigger, assuming we have stable, pubsub peers in good case

Reply:
We can resolve @nashqueue's issue here by making the multiplier configurable, so every network diced on their own

Reply:
I will just make the recencyThreshold configurable and set the default to `blockTime+5 seconds`

Reply:
Also fine